src/main/java/com/xceptance/xlt/agentcontroller/FileManagerProxy.java

```diff
@@ -21,44 +21,94 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.io.RandomAccessFile;
 import java.net.HttpURLConnection;
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.net.URLConnection;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.IOUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.xceptance.common.io.IoActionHandler;
+import com.xceptance.common.lang.ParseNumbers;
 import com.xceptance.common.net.UrlConnectionFactory;
+import com.xceptance.xlt.api.util.XltException;
+import com.xceptance.xlt.engine.httprequest.HttpRequestHeaders;
+import com.xceptance.xlt.engine.httprequest.HttpResponseHeaders;
 
 /**
  * The FileManagerProxy class is the client-side implementation of the FileManager interface, i.e. it runs on the master
  * controller.
  */
 public class FileManagerProxy implements FileManager
 {
+    /**
+     * The result returned when downloading a chunk.
+     */
+    private static class ChunkInfo
+    {
+        /** The total size of the resource that is currently being downloaded. */
+        public final long totalSize;
+
+        /** The size of the current chunk. */
+        public final long chunkSize;
+
+        public ChunkInfo(final long totalSize, final long chunkSize)
+        {
+            super();
+            this.totalSize = totalSize;
+            this.chunkSize = chunkSize;
+        }
+    }
+
     private static final Logger log = LoggerFactory.getLogger(FileManagerProxy.class);
 
+    /**
+     * A pattern to validate a Content-Range response header (for example, <code>bytes 1000-1999/12345</code>) and
+     * extract values from it.
+     */
+    private static final Pattern CONTENT_RANGE_PATTERN = Pattern.compile("bytes (\\d+)-(\\d+)/(\\d+)");
+
```

Comment:
Potential improvement: Parsing and formatting of range headers might be moved to some utility class (accompanied w/ some tests if feasible).

Reply:
Done.